src/main/java/io/horrorshow/codey/time/DiscordTimezone.java

```diff
@@ -0,0 +1,135 @@
+package io.horrorshow.codey.time;
+
+import io.horrorshow.codey.discordutil.DiscordUtils;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+
+@Service
+public class DiscordTimezone extends ListenerAdapter {
+    private final DiscordUtils utils;
+
+    public DiscordTimezone(@Autowired JDA jda, @Autowired DiscordUtils utils) {
+        this.utils = utils;
+
+        jda.addEventListener(this);
+    }
+
+    private final String emoji = "⏰";
+    private final String deleteEmoji = "\uD83D\uDDD1";
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        if (!event.getAuthor().isBot()) {
+            onMessage(event.getMessage());
+        }
+    }
+
+    private void onMessage(Message message) {
+        String rawContent = message.getContentRaw();
+        String[] rawContentParts = rawContent.split("\\s+");
+
+        if(rawContentParts.length == 0)
+            return;
+
+        Object[] result = time(rawContentParts);
+
+        if(!((boolean) result[0])) {
+            if(result[1] != null && (boolean) result[1])
+            {
+                message.addReaction(emoji).complete();
+                message.reply(new EmbedBuilder()
+                        .setColor(utils.getColor())
+                        .setTitle("Did you know that I can provide time conversions automatically?")
+                        .addField("Here is a tip:",
+                                "You can use regions from the IANA Time Zone Database (TZDB). " +
+                                "This has region IDs of the form `{area}/{city}`, such as `Europe/Paris` or `America/New_York`", false)
+                        .build()).complete().addReaction(deleteEmoji).complete();
+            }
+            return;
```

Comment:
instead of returning here the method would be easier to read looking like (given that it works with CompletableFutures):
```java
findTime(rawContent).thenApply(timeAsString -> toJavaDateTime(timeAsString)).thenAccept(dateTime -> send the discord message...);
```
But we'll look into that once the `time()` method is cleaned up and stable.